ios/PurchasesHybridCommon/PurchasesHybridCommonSwift/CommonFunctionality.swift

```diff
@@ -0,0 +1,505 @@
+//
+//  CommonFunctionality.swift
+//  PurchasesHybridCommonSwift
+//
+//  Created by Andrés Boedo on 4/19/22.
+//  Copyright © 2022 RevenueCat. All rights reserved.
+//
+
+import Foundation
+import StoreKit
+import Purchases
+
+
+// todo: rename back to RCCommonFunctionality
+@objc(RCCommonFunctionality) public class CommonFunctionality: NSObject {
+
+    // MARK: properties and configuration
+    @objc public static var simulatesAskToBuyInSandbox: Bool = false
+    @objc public static var appUserID: String { Purchases.shared.appUserID }
+    @objc public static var isAnonymous: Bool { Purchases.shared.isAnonymous }
+
+    @objc public static var proxyURLString: String? {
+        get { Purchases.proxyURL?.absoluteString }
+        set {
+            if let value = newValue {
+                guard let proxyURL = URL(string: value) else {
+                    fatalError("could not set the proxyURL, provided value is not a valid URL: \(value)")
+                }
+                Purchases.proxyURL = proxyURL
+            } else {
+                Purchases.proxyURL = nil
+            }
+        }
+    }
+
+    @objc public var simulatesAskToBuyInSandbox: Bool {
+        get {
+            // all other platforms already support this feature
+            if #available(macOS 10.14, *) {
+                return Purchases.simulatesAskToBuyInSandbox
+            } else {
+                return false
+            }
+        }
+        set {
+            // all other platforms already support this feature
+            if #available(macOS 10.14, *) {
+                Purchases.simulatesAskToBuyInSandbox = newValue
+            } else {
+                NSLog("called setSimulatesAskToBuyInSandbox, but it's not available on this platform / OS version")
+            }
+        }
+    }
+
+    private static var _discountsByProductIdentifier: Any? = nil
+
+    @available(iOS 12.2, macOS 10.14.4, tvOS 12.2, *)
+    private static var discountsByProductIdentifier: [String: SKPaymentDiscount] {
+        get {
+            return _discountsByProductIdentifier as! [String: SKPaymentDiscount]
+        } set {
+            _discountsByProductIdentifier = newValue
+        }
+    }
+
+    @objc public static func configure() {
+        // todo: it seems like this call isn't needed anymore?
+    }
```

Comment:
looks like we should be able to kill this one. I'm keeping it until right _after_ the migration just in case

Reply:
the only use for this one before was to initialize a static variable, but in Swift we can do that in its declaration